### PR TITLE
Final group capstone - Book an Appointment

### PR DESCRIPTION
**Pull request by yigitm**
In this PR:

- React app is created with redux & thunk & jest & tailwind
- Add/Delete/List endpoints are consumed via Rails API.
- Basic authentication is created using client session storage 
- Styling is based on the design mentioned in the instructions

[Project Requirements](https://github.com/microverseinc/curriculum-final-capstone/blob/main/projects/business_requirements.md)

**After running 'heroku run rails db:migrate' command; I got the below error. That's why for now on; I can only provide the front end's live version link. 

heroku-error

**If you run the Rails API in your localhost/3000; you can test the [Live version of Front End](https://nimble-hummingbird-7623a3.netlify.app/).

[Latest Branch For Back End Repository](https://github.com/yigitm/book-appointment/tree/API-readme)  

** I would be grateful if you may have any suggestions to fix this error. Thank you for your time.

### CODE REVIEW

File: src/components/mobile/CoursesMob.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+          alt="/"
+        />
+      );
+    }
+    if (item.course_type === 'Cabin Course') {
+      return (
+        <img
+          className="w-[200px] mx-auto bg-cyan-100 rounded-full"
+          src={cabin}
+          alt="/"
```
Comment (mcrd25) [inline src/components/mobile/CoursesMob.js]: [OPTIONAL] Maybe this img can be a reusable component: You judge for yourself

File: src/components/home/Courses.js [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+              {state.length === 0 ? (
+                <div>
+                  <img className="w-full" src={plane} alt="/" />
+                  <p className="font-thin text-center mb-20">
+                    No course added, yet.
+                  </p>
+                </div>
+              ) : null}
+              {sessionStorage.getItem('TOKEN') ? null : (
+                <div>
```
Comment (mcrd25) [inline src/components/home/Courses.js]: There are some candidates to make some of these html components reusable.

**Review by mcrd25 — CHANGES_REQUESTED** [review]
Hi @yigitm >, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!



## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- The issue you are having on Heroku probably has to do with where the `rwag` gem is located in gem file, rather the group it's located in - See this thread: https://github.com/rswag/rswag/issues/261
-  This front-end definitely has to look closer to he design - Remember you are a developer, not the designer. In real world the designer already has done design that client has approved of, it is your job to have it match as much as possible. I believe with a bit of effort you can get there 💪 
- You definitely have some candidates for resusable components
- I've tried to be fair but may have missed some areas of improvement, please see where you can improve your code and do things better

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

*You are also free to follow me on github and socials 😄 *

File: src/components/mobile/CoursesMob.js (hunk 1 above)
Comment (yigitm) [reply src/components/mobile/CoursesMob.js]: Thank you. Refactored both desktop & mobile components.

File: src/components/home/Courses.js (hunk 2 above)
Comment (yigitm) [reply src/components/home/Courses.js]: Thank you. Refactored both desktop & mobile components.

**Pull request merged**